### ISSUE TITLE
[FEATURE REQUEST] Replace NestJS Logger Entirely

### ISSUE THREAD

**Issue by gf3** [iamolegga/nestjs-pino#477]
**Is your feature request related to a problem? Please describe.**
The current method of replacing the NestJS logger means we miss out on some important logs during app boot, or we allow the default logger for boot which doesn't match our format (or get sent to our other log streams if using pino-multistream).

**Describe the solution you'd like**
nest-winston has [a method](https://github.com/gremo/nest-winston/blob/0a1cacd901378c1692eb6094aa707f000726dcb9/src/winston.module.ts#L29-L31) which is [described in their documentation](https://github.com/gremo/nest-winston#replacing-the-nest-logger-also-for-bootstrapping) to replace the app logger at boot time; is it possible to use a similar strategy to replace the default NestJS logger even earlier?

**Additional context**
Both log formatting and multiple streams are important to us here, so ideally we'd like to continue to use our singular configured logger globally and still catch critical boot messages.

**Comment by iamolegga** [comment]
Good point. Need time to investigate this.

Also, just fyi in new major version of nest there should be buffering of this kind of messages, so loggers with async initialisation should handle this too. But that's the future, and it will require some time to implement new logger api for v8

**Comment by ccontour** [comment]
I use this method to get into nest at boot time in my code. Little roundabout, but it gets all the module init messages

```
import { Logger, PinoLogger } from 'nestjs-pino';

export const BootstrapLogger = (level = 'debug'): { logger: Logger } => {
  return {
    logger: new Logger(
      new PinoLogger({
        pinoHttp: {
          level,
        },
      }),
      {
        pinoHttp: {
          level,
        },
      },
    ),
  };
};
```

```
  const app = await NestFactory.create(AppModule, {
    ...BootstrapLogger(),
    cors: true,
  });
```

**Comment by gf3** [comment]
thanks for sharing @ccontour

**Comment by iamolegga** [comment]
solved in v2.0.0
